docs/vm-configuration.md

```diff
@@ -264,3 +264,95 @@ the generation of multiple libvirt XML documents, the domain XML and associated
 documents for secrets, network filters and node devices. So there is not a strict
 1-1 mapping between the k8s API resource representation and the libvirt low
 level representation.
+
+Proposed VM Definition
+----------------------
+
+The following definition adheres to main points described above. It provides
+
+1. high level description usable by end users or management applications and
+2. low level details required to construct and launch a VM.
+
+The features section is used to generate the domain section, the other way
+around is out of scope for this proposal. Each feature must represent one or
+more fields in the domain section.
+
+The design and contents of each feature section should be logically split and
+discussed in separate proposal. This is just an example without the sections
+discussed, converted from cluster/vm.yaml. Splitting the features enables us to
+create modular VM Spec generation flows that are easily tunable by experts in
+their domains of expertise.
+
+```yaml
+apiVersion: kubevirt.io/v1alpha2
+kind: VM
+metadata:
+  name: testvm
+spec:
+  features:
```

Comment:
@fabiand,@rmohr: How about looking at it as:

kind: VMConfig
spec:
  features:
    .. # as described in this proposal

# THIS IS THEN TRANSFORMED on /start to a VM OBJ
kind: VM
  features:
    .. # as described in this proposal
  domain:
    .. # resulting detailed structure, consumed by virt-handler

- domain resembles libvirt XML in structure (all detailed devices, PCI addresses?, timers, etc), it is further transformed by virt-handler to resulting libvirt XML, but that expansion is only simple straightforward one (like adding kubevirt-hardcoded options, e.g. "os type hvm", etc) and plus all the host-specific things (actual mapping of the backend resources)
- the domain part is "generated" by the admission controller of the VM entity, that's where all the logic of how the VM looks like resides, based on e.g. OS and features like "use virtio-scsi".
- there is a duplicity between features and domain section in VM. You can basically either post only "features"(high level) or only "domain"(low level description) or both, but in that case you need to know about potential conflicts in implementation of each "feature->domain" in the admission controller. Still, it seems to me as the lesser of all other evils.

doesn't it solve the problem of the main entry point? This way interacting with VM still gives you full power and the logic is kept in single place, VMConfig is only for persistence, separate from runtime.